B/boost/build_tarballs.jl

```diff
@@ -42,15 +46,18 @@ if [[ $target == *apple* ]]; then
     fi
 elif [[ $target == x86_64*mingw* ]]; then
     targetos=windows
-    extraargs="address-model=64 binary-format=pe abi=ms link=shared"
+    extraargs="address-model=64 define=_WIN32_WINNT=0x0602 binary-format=pe abi=ms link=shared"
```

Comment:
Having to define `_WIN32_WINNT=0x0602` is surprising, since we already define [`_WIN32_WINNT=0x0A00`](https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/a79769dc67d3adcd1ab211b5e29a489a6e46841c/src/Runner.jl#L520) in our compiler wrappers and `0x0A00 > 0x0602` ([`0x0602` is Windows 8, `0x0A00` is Windows 10](https://learn.microsoft.com/en-us/cpp/porting/modifying-winver-and-win32-winnt?view=msvc-170))

Reply:
It's the Windows 10 setting that wasn't working, and switching to Windows 8 makes it work. This function from `winapi/handles.hpp` isn't found (at link time IIRC):
```
#if BOOST_USE_WINAPI_VERSION >= BOOST_WINAPI_VERSION_WIN10
BOOST_WINAPI_IMPORT boost::winapi::BOOL_ BOOST_WINAPI_WINAPI_CC
CompareObjectHandles(
    boost::winapi::HANDLE_ hFirstObjectHandle,
    boost::winapi::HANDLE_ hSecondObjectHandle);
#endif
```